### PR TITLE
Fixes deprecation in NWListener.swift

### PR DESCRIPTION
**Pull request by valeriyvan**
Fixes deprecation in NWListener.swift

### CODE REVIEW

**Comment by xwu** [comment]
My understanding is that framework overlays have even returned to their respective framework owners; this code is vestigial.

(Also, you have disturbed the indenting.)

**Comment by CodaFi** [comment]
@xwu My comments were a little too aggressive. What I meant at the time was that we in the Swift project need to go ask the right sets of code owners for their permission to merge. We can still keep hacking on overlays!

**Comment by xwu** [comment]
@CodaFi Oh yay! Glad to hear.

**Comment by CodaFi** [comment]
rdar://63724290

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: stdlib/public/Darwin/Network/NWListener.swift
Comment (CodaFi) [inline stdlib/public/Darwin/Network/NWListener.swift:82]: Nit: As was said before, we ought to get this corrected before we merge.

Comment (lorentey) [reply stdlib/public/Darwin/Network/NWListener.swift:82]: The type annotations for the closure were only needed to call the deprecated variant, so they ought to be removed. With the buffer pointer-based entry points, we prefer to extract the count from the buffer pointer itself, not the original class instance:

```swift
txtRecord.withUnsafeBytes { buffer in
 nw_advertise_descriptor_set_txt_record(descriptor!, buffer.baseAddress!, buffer.count)
}
```

Comment (valeriyvan) [reply stdlib/public/Darwin/Network/NWListener.swift]: fixed

**Review by CodaFi — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by valeriyvan** [comment]
ping

**Comment by valeriyvan** [comment]
Network was removed from stdlib

**Pull request closed without merging**
